test/repeater-test.js

```diff
@@ -166,6 +166,36 @@ define(function(require){
 		});
 	});
 
+	test("should handle canceling search correctly", function () {
+		var count = -1;
+		var $repeater = $(this.$markup);
+		var $search = $repeater.find('.repeater-search');
+		$repeater.repeater({
+			dataSource: function(options, callback){
+				count++;
+				switch (count){
+					case 0:
+						equal(options.search, undefined, 'search value not passed to dataSource initially as expected');
+						callback({});
+						$search.find('input').val('something');
+						$search.trigger('searched.fu.repeater');
+						break;
```

Comment:
Yep, this definitely doesn't work. I think I remembered figuring out why at some point in the past and it affected all these tests.  Partially its that the tests needs to run async using https://api.qunitjs.com/async/ and not to signal the test is over until all assertions have completed. Because triggering these DOM events is not guaranteed to be synchronous operation in every browser. I tried adding an async in here and the 2nd & 3rd branches still don't work, I'll mess with it more tomorrow morning.